lib/functions/compilation/kernel-config.sh

```diff
@@ -138,3 +149,25 @@ function kernel_config_export() {
 		run_host_command_logged cp -pv defconfig "${kernel_config_source_filename}.defconfig"
 	fi
 }
+
+# Manually forcing kernel options with 'call_extensions_kernel_config()' can introduce missing dependencies or misconfigurations
+# This function checks the config, re-establishes its sanity if necessary and outputs any changes to the user
+function kernel_config_check_and_repair() {
+	# Re-run kernel make to automatically solve any dependencies and/or misconfigurations
+	run_kernel_make_dialog oldconfig
```

Comment:
I would suggest to use olddefconfig instead.

Reply:
IIRC, in cases like rewrite-kernel-config and even during normal kernel builds using kernel command, the build is generally run as non-interactive process and any user input are ignored and not passed to the subprocess. In cases when oldconfig would require user input the process would just hang waiting for user input. Hence I suggested to go with olddefconfig